rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,165 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+	var scaleFactor float64
+	if (maxX-minX) != 0 || (maxZ-minZ) != 0 {
+		widthScaleFactor := float64(imageWidth-1) / (maxX - minX)
+		heightScaleFactor := float64(imageHeight-1) / (maxZ - minZ)
+		scaleFactor = math.Min(widthScaleFactor, heightScaleFactor)
+	}
+
+	// Add points in the pointcloud to a new image
+	var voxelColor rimage.Color
+	im := rimage.NewImage(imageWidth, imageHeight)
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		x := int(math.Round((pt.X - minX) * scaleFactor))
+		y := int(math.Round((pt.Z - minZ) * scaleFactor))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// the default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight && data != nil {
+			voxelColor, err = getProbabilityColorFromValue(data)
+			if err != nil {
+				return false
+			}
+			im.Circle(image.Point{X: x, Y: y}, pointRadius, voxelColor)
+		}
+		return true
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Add a red robot marker to the image
+	if ppRM.robotPose != nil {
+		x := int(math.Round((robotMarker.Point().X - minX) * scaleFactor))
+		y := int(math.Round((robotMarker.Point().Z - minZ) * scaleFactor))
+		robotMarkerColor := rimage.NewColor(255, 0, 0)
+		im.Circle(image.Point{X: x, Y: y}, robotMarkerRadius, robotMarkerColor)
+	}
+	return im, nil, nil
+}
+
+// RGBDToPointCloud is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
+	img *rimage.Image,
+	dm *rimage.DepthMap,
+	crop ...image.Rectangle,
+) (pointcloud.PointCloud, error) {
+	return nil, errors.New("converting an RGB image to Pointcloud is currently unimplemented for this projection")
+}
+
+// ImagePointTo3DPoint is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
+	return r3.Vector{}, errors.New("converting an image point to a 3D point is currently unimplemented for this projection")
+}
+
+// getProbabilityColorFromValue returns an RGB color value based on the probability value and defined hit and miss
+// thresholds
+// TODO (RSDK-1705): Once probability values are available, a temporary algorithm is being used based on Cartographer's method
+// of painting images.
+func getProbabilityColorFromValue(d pointcloud.Data) (rimage.Color, error) {
+	var r, g, b uint8
+
+	if !d.HasValue() {
+		return rimage.NewColor(255, 255, 255), nil
+	}
+
+	if d.Value() > 100 || d.Value() < 0 {
+		return rimage.NewColor(0, 0, 0),
+			errors.Errorf("received a value of %v which is outside the range (0 - 100) representing probabilities", d.Value())
+	}
+
+	prob := float64(d.Value()) / 100.
+
+	switch {
+	case prob < missThreshold:
+		b = uint8(255 * (prob / missThreshold))
+	case prob > hitThreshold:
+		g = uint8(255 * ((prob - hitThreshold) / (1 - hitThreshold)))
+	default:
+		r, b, g = 255, 255, 255
```

Comment:
Won't this make it so that values lower than `missThreshold` will apply a gradient, values above the `hitThreshold` will also provide a different gradient, but values in-between will be solid white?

I would expect `solid white` to represent certainty. Am I misunderstanding how this will behave or the meaning of the colors?

Reply:
Fixed!